__tests__/app.test.js

```diff
@@ -31,9 +33,560 @@ describe("NCNews-Server Unit Tests", () => {
                 });
             });
         });
-       
-    
+   
+    });
+
+    describe(`GET /api/articles/`, () => {
+                
+        test(`[ 200 ] Responds with an array of objects, where the objects have the following 7 properties: author, title, article_id, topic, created_at, votes, comment_count`, () => {
+
+            return request(app).get(`/api/articles`).expect(200).then((response) => {
+                const articlesArray = response.body.articles;
+                expect(articlesArray).toHaveLength(12);
+
+                articlesArray.forEach((articleObject) => {
+                    expect(articleObject).toEqual(
+                        expect.objectContaining({
+                            author: expect.any(String),
+                            title: expect.any(String),
+                            topic: expect.any(String),
+                            article_id: expect.any(Number),
+                            created_at: expect.any(String),
+                            votes: expect.any(Number),
+                            comment_count: expect.any(Number)
+                    }));
+                });
+            });
+        });
+
+        test(`[ 200 ] Responds with an array of objects, where the objects are sorted by Date using the 'created_at' property in Descending order (latest dates first)`, () => {
+
+            return request(app).get(`/api/articles`).expect(200).then((response) => {
+                const articlesArray = response.body.articles;   
+                expect(articlesArray).toHaveLength(12);
+
+                articlesArray.forEach((e, index) => {
+                    if (index < articlesArray.length - 1) {
+                    expect(Date.parse(articlesArray[index][`created_at`]) > Date.parse(articlesArray[index + 1][`created_at`])).toBe(true);
+                }});
+            });
+
+        });
+
+        // Task 10 - add in queries
+
+        describe(`GET /api/articles - URL Parameters`, () => {
+
+            // include test to make sure it works for articles? aswell as articles/?
+            // include thorough sql injection tests in each section below
+
+            describe(`GET /api/articles/?topic=`, () => {
+
+                test(`[ 200 ] When passed "topic=cats" as a url parameter, responds with the 1 cat topic article`, () => {
+
+                    return request(app).get(`/api/articles?topic=cats`).expect(200).then((response) => {
+                        const articlesArray = response.body.articles;   
+                        expect(articlesArray).toHaveLength(1);
+                        expect(articlesArray[0]["topic"]).toBe("cats");
+        
+                    });
+                });
+
+                test(`[ 200 ] A topic with zero matches returns an empty array`, () => {
+
+                    return request(app).get(`/api/articles?topic=k`).expect(200).then((response) => {
```

Comment:
I dont think this one is necessarily a 200, it could be a 404 because this is a non-existent category rather than a category that exists but doesnt have any comments yet